endpoint.cpp

```diff
@@ -87,14 +89,27 @@ Endpoint::Endpoint(const char *name, bool crc_check_enabled)
 
 Endpoint::~Endpoint()
 {
-    if (fd >= 0) {
-        ::close(fd);
-    }
-
     free(rx_buf.data);
     free(tx_buf.data);
 }
 
+bool Endpoint::handle_canwrite()
+{
+    int r = flush_pending_msgs();
+    return r == -EAGAIN;
+}
+
+void Endpoint::handle_read()
+{
+    assert(_router);
+
+    int target_sysid;
+    struct buffer buf{};
+
+    if (read_msg(&buf, &target_sysid) > 0)
+        _router->router_msg(&buf, target_sysid, _system_id);
```

Comment:
in the Endpoint we relied on read_msg() being called on a loop:

```
-    while (endpoint->read_msg(&buf, &target_sysid) > 0)
-        router_msg(endpoint, &buf, target_sysid);
-}
```

why did you change it to one read per loop iteration?